forta-bots/L2-DAI-Monitor/src/agent.ts

```diff
@@ -0,0 +1,40 @@
+import { BlockEvent, Finding, FindingSeverity, FindingType, getEthersProvider } from "forta-agent";
+import abi from "./abi";
+import { ethers, BigNumber, Contract } from "ethers";
+
+const DAI: string = "0xDA10009cBd5D07dd0CeCc66161FC93D7c9000da1";
+
+export const provideHandleBlock = (provider: ethers.providers.JsonRpcProvider, dai: string) => {
+  let balance: BigNumber = BigNumber.from(-1);
+  const daiContract: Contract = new Contract(dai, abi.DAI, provider);
+
+  return async (blockEvent: BlockEvent) => {
+    const findings: Finding[] = [];
+
+    const currentBalance: BigNumber = await daiContract.totalSupply({ blockTag: blockEvent.blockNumber });
+    if (!balance.eq(currentBalance)) {
+      findings.push(
+        Finding.from({
+          alertId: "L2-DAI-MONITOR",
+          description: "Balance change detected",
```

Comment:
Here and in the "name" field I think it would be better to use "Total supply" instead of "Balance"